File: backend/src/main/java/com/team01/backend/domain/notification/repository/SseEmitterRepository.java
package com.team01.backend.domain.notification.repository;

import org.springframework.stereotype.Repository;
import org.springframework.web.servlet.mvc.method.annotation.SseEmitter;

import java.util.List;
import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;
import java.util.concurrent.CopyOnWriteArrayList;

@Repository
public class SseEmitterRepository {
    private final Map<Long, List<SseEmitter>> emitters = new ConcurrentHashMap<>();

    public void save(Long userId, SseEmitter emitter) {
        emitters.computeIfAbsent(userId, k -> new CopyOnWriteArrayList())
                .add(emitter);
    }

    public List<SseEmitter> findByUserId(Long userId) {
        return emitters.getOrDefault(userId, List.of());
    }

    public void delete(Long userId, SseEmitter emitter) {
        List<SseEmitter> list = emitters.get(userId);
        if (list != null) {
            list.remove(emitter);
        }
    }
}

File: backend/src/main/java/com/team01/backend/domain/notification/repository/SseEmitterRepository.kt
package com.team01.backend.domain.notification.repository

import org.springframework.stereotype.Repository
import org.springframework.web.servlet.mvc.method.annotation.SseEmitter
import java.util.concurrent.ConcurrentHashMap
import java.util.concurrent.CopyOnWriteArrayList

@Repository
class SseEmitterRepository {
    private val emitters = ConcurrentHashMap<Long, MutableList<SseEmitter>>()

    fun save(userId: Long, emitter: SseEmitter) {
        emitters.getOrPut(userId){CopyOnWriteArrayList()}
            .add(emitter)
    }

    fun findByUserId(userId: Long): List<SseEmitter> {
        return emitters[userId]?: emptyList()
    }

    fun delete(userId: Long, emitter: SseEmitter) {
        emitters[userId]?.remove(emitter)
    }
}